gvm/protocols/gmpv7.py

```diff
@@ -1935,40 +1935,290 @@ def modify_filter(self, filter_id, comment=None, name=None, term=None,
+                raise InvalidArgument('modify_permission requires a subject '
+                                      'dict with id and type members')
+            _xmlsubject = cmd.add_element('subject', attrs={'id': subject_id})
+            _xmlsubject.add_element('type', subject_type)
+
+        return self._send_xml_command(cmd)
+
+    def modify_port_list(self, port_list_id, comment=None, name=None, ):
+        """Generates xml string for modify port list on gvmd.
+
+        Arguments:
+            port_list_id (str): UUID of port list to modify.
+            name (str, optional): Name of port list.
+            comment (str, optional): Comment on port list.
+        """
+        if not port_list_id:
+            raise RequiredArgument('modify_port_list requires '
+                                   'a port_list_id attribute')
+        cmd = XmlCommand('modify_port_list')
+        cmd.set_attribute('port_list_id', port_list_id)
+
+        if comment:
+            cmd.add_element('comment', comment)
+
+        if name:
+            cmd.add_element('name', name)
+
+        return self._send_xml_command(cmd)
 
     def modify_report(self, report_id, comment):
-        cmd = self._generator.modify_report_format_command(report_id, comment)
-        return self.send_command(cmd)
+        """Generates xml string for modify report on gvmd.
 
-    def modify_report_format(self, report_format_id, **kwargs):
-        cmd = self._generator.modify_report_format_command(report_format_id,
-                                                           kwargs)
-        return self.send_command(cmd)
+        Arguments:
+            report_id (str): UUID of report to modify.
+            comment (str): The comment on the report.
+        """
+        if not report_id:
+            raise RequiredArgument('modify_report requires '
+                                   'a report_id attribute')
+        if not comment:
+            raise RequiredArgument('modify_report requires '
+                                   'a comment attribute')
+        cmd = XmlCommand('modify_report')
+        cmd.set_attribute('report_id', report_id)
+        cmd.add_element('comment', comment)
 
-    def modify_role(self, role_id, **kwargs):
-        cmd = self._generator.modify_role_command(role_id, kwargs)
-        return self.send_command(cmd)
+        return self._send_xml_command(cmd)
+
+    def modify_report_format(self, report_format_id, active=None, name=None,
+                             summary=None, param=None):
+        """Generates xml string for modify report format on gvmd.
+
+        Arguments:
+            report_format_id (str) UUID of report format to modify.
+            active (boolean, optional): Whether the report format is active.
+            name (str, optional): The name of the report format.
+            summary (str, optional): A summary of the report format.
+            param (list, optional): List members: [name, value]]: The name of
```

Comment:
Shouldn't it be params as the argument name?

Wouldn't it be better to use a dict for param here? Or can the same name be listed more the once?

Reply:
Looking at the protocol docs it seems that only one name and only one value can be passed. So param should be converted into two string parameters namely param_name and param_value